WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -2421,7 +2421,7 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
                     // user chose a featured image
                     if (data.hasExtra(MediaPickerConstants.EXTRA_MEDIA_ID)) {
                         long mediaId = data.getLongExtra(MediaPickerConstants.EXTRA_MEDIA_ID, 0);
-                        setFeaturedImageId(mediaId);
+                        setFeaturedImageId(mediaId, true);
```

Comment:
Probably I'm just missing something stupid but is this case for future steps we still need to cover? it seems I cannot reach here with the current capabilities we have for setting the featured image 🤔

Reply:
I'd expect this place to be the one that covers the featured image selection (since it's the only place where `imagePicked` is true and I saw the `IMAGE_PICKED` event in the logs). When checking this I noticed that we need to somehow cover the featured image selection options that we currently have (it allows the user to select an image from all the options like stock library and gifs).

Reply:
In the case of flag off (like in Vanilla) I think we trigger the tracking event two times if we leave the `setFeaturedImageId`to true. I know we will need to review the tracking for this feature overall, but what if we replace the true with a check on if the flag is enabled? wdyt?

Reply:
that sounds correct! good catch